perception/util/visualizer/traffic_light_visualization/src/nodelet.cpp

```diff
@@ -18,43 +18,52 @@
 
 namespace traffic_light
 {
-void TrafficLightRoiVisualizerNodelet::onInit()
-{
-  nh_ = getNodeHandle();
-  pnh_ = getPrivateNodeHandle();
-  image_transport_.reset(new image_transport::ImageTransport(pnh_));
-
-  pnh_.param<bool>("enable_fine_detection", enable_fine_detection_, false);
+TrafficLightRoiVisualizerNodelet::TrafficLightRoiVisualizerNodelet(const rclcpp::NodeOptions & options)
+: Node("traffic_light_roi_visualizer_node", options)
+{  
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+  using std::placeholders::_3;
+  enable_fine_detection_ = this->declare_parameter("enable_fine_detection", false);
 
   if (enable_fine_detection_) {
     sync_with_rough_roi_.reset(new SyncWithRoughRoi(SyncPolicyWithRoughRoi(10), image_sub_, roi_sub_,  rough_roi_sub_));
-    sync_with_rough_roi_->registerCallback(boost::bind(&TrafficLightRoiVisualizerNodelet::imageRoughRoiCallback, this, _1, _2, _3));
+    sync_with_rough_roi_->registerCallback(std::bind(&TrafficLightRoiVisualizerNodelet::imageRoughRoiCallback, this, _1, _2, _3));
   } else {
     sync_.reset(new Sync(SyncPolicy(10), image_sub_, roi_sub_));
-    sync_->registerCallback(boost::bind(&TrafficLightRoiVisualizerNodelet::imageRoiCallback, this, _1, _2));
+    sync_->registerCallback(std::bind(&TrafficLightRoiVisualizerNodelet::imageRoiCallback, this, _1, _2));
   }
-  image_transport::SubscriberStatusCallback connect_cb = boost::bind(&TrafficLightRoiVisualizerNodelet::connectCb, this);
-  std::lock_guard<std::mutex> lock(connect_mutex_);
```

Comment:
What was the rationale behind removing the mutex here and in the method `connectCb()`? Also I noticed that the `connect_mutex_` member variable is still in the header but not used

Reply:
the declaration in the header was missed, I've removed it now. 
I don't think the mutex is needed anymore, if you see in the old code the publisher `image_pub_` needed to use `connect_cb` to `advertise` with `connect_cb` also accessing `image_pub_`, but that is not happening anymore as `image_pub_` is not dependent on `connect_cb`

Reply:
I see, thanks for the explanation!